scene/gui/texture_button.cpp

```diff
@@ -343,6 +344,27 @@ void TextureButton::set_focused_texture(const Ref<Texture> &p_focused) {
 	focused = p_focused;
 };
 
+void TextureButton::_set_texture(Ref<Texture> *p_destination, const Ref<Texture> &p_texture) {
+	DEV_ASSERT(p_destination);
```

Comment:
`DEV_ASSERT` should only be used for bottleneck code. A regular check here won't matter.

But I think a better way is to pass by non-const reference, so we don't have to do any check.

Reply:
See: https://github.com/godotengine/godot/pull/77159#discussion_r1209237736

Reply:
> That won't work, assigning to the reference won't update the texture itself

The referenced `Ref` object won't change. What will be updated is the content of that `Ref`, which is the texture.

> Also `DEV_ASSERT` is for crashing, not performance critical AFAIK

A note was added recently:
https://github.com/godotengine/godot/blob/8449592d92eaeef990f5502b419d491ee3eeb7a6/core/error/error_macros.h#L797-L801

But anyway, I think it's fine if the Godot 4 version uses this. I should have read the original PR, sorry :(